src/platform_impl/windows/keyboard.rs

```diff
@@ -0,0 +1,665 @@
+                        next_msg.as_mut_ptr(),
+                        hwnd,
+                        winuser::WM_KEYFIRST,
+                        winuser::WM_KEYLAST,
+                        winuser::PM_NOREMOVE,
+                    );
+                    let has_message = has_message != 0;
+                    if !has_message {
+                        more_char_coming = false;
+                    } else {
+                        let next_msg = next_msg.assume_init().message;
+                        if next_msg == winuser::WM_CHAR || next_msg == winuser::WM_SYSCHAR {
+                            more_char_coming = true;
+                        } else {
+                            more_char_coming = false;
+                        }
+                    }
+                }
+
+                if is_utf16 {
+                    self.event_info
+                        .as_mut()
+                        .unwrap()
+                        .utf16parts
+                        .push(wparam as u16);
+                } else {
+                    // Otherwise wparam holds a utf32 character.
+                    // Let's encode it as utf16 appending it to the end of utf16parts
+                    let utf16parts = &mut self.event_info.as_mut().unwrap().utf16parts;
+                    let start_offset = utf16parts.len();
+                    let new_size = utf16parts.len() + 2;
+                    utf16parts.resize(new_size, 0);
+                    if let Some(ch) = char::from_u32(wparam as u32) {
+                        let encode_len = ch.encode_utf16(&mut utf16parts[start_offset..]).len();
+                        let new_size = start_offset + encode_len;
+                        utf16parts.resize(new_size, 0);
+                    }
+                }
+                if !more_char_coming {
+                    let mut event_info = self.event_info.take().unwrap();
+
+                    let mut layouts = LAYOUT_CACHE.lock().unwrap();
+                    // Here it's okay to call `ToUnicode` because at this point the dead key
+                    // is already consumed by the character.
+                    let kbd_state = get_kbd_state();
+                    let mod_state = WindowsModifiers::active_modifiers(&kbd_state);
+
+                    let (_, layout) = layouts.get_current_layout();
+                    let ctrl_on;
+                    if layout.has_alt_graph {
+                        let alt_on = mod_state.contains(WindowsModifiers::ALT);
+                        ctrl_on = !alt_on && mod_state.contains(WindowsModifiers::CONTROL)
+                    } else {
+                        ctrl_on = mod_state.contains(WindowsModifiers::CONTROL)
+                    }
+
+                    // If CTRL is not pressed, just use the text with all
+                    // modifiers because that already consumed the dead key and otherwise
+                    // we would interpret the character incorretly, missing the dead key.
```

Comment:
```suggestion
                    // If Ctrl is not pressed, just use the text with all
                    // modifiers because that already consumed the dead key. Otherwise,
                    // we would interpret the character incorrectly, missing the dead key.
```